packages/indexer-agent/src/network.ts

```diff
@@ -120,16 +133,134 @@ export class Network {
       return 'unauthorized'
     }
 
+    let pending = true
+    let output: providers.TransactionReceipt | undefined = undefined
+
     const estimatedGas = await gasEstimation()
-    const tx = await transaction(Math.ceil(estimatedGas.toNumber() * 1.5))
-    logger.info(`Transaction pending`, { tx: tx.hash })
-    const receipt = await tx.wait(2)
-    logger.info(`Transaction successfully included in block`, {
-      tx: tx.hash,
-      blockNumber: receipt.blockNumber,
-      blockHash: receipt.blockHash,
+    const txPromise = transaction(Math.ceil(estimatedGas.toNumber() * 1.5))
+    let tx = await txPromise
+
+    let txConfig: Record<string, number> = {
+      attempt: 1,
+      nonceOffset: 0,
+      gasBump: 1 + this.gasBumpPercent / 100,
+      nonce: tx.nonce,
+      gasPrice: tx.gasPrice.toNumber(),
+      gasLimit: tx.gasLimit.toNumber(),
+    }
+
+    logger.info(`Sending transaction`, { tx: tx, attempt: txConfig.attempt })
+
+    while (pending) {
+      if (txConfig.attempt >= this.maxRetries) {
+        logger.warn('Transaction attempt threshold surpassed, moving on', {
+          attempts: txConfig.attempt,
+        })
+        break
+      }
+
+      try {
+        if (txConfig.attempt > 1) {
+          logger.info('Resubmitting transaction', {
+            txConfig: txConfig,
+          })
+          const transaction: providers.TransactionRequest = {
+            value: tx.value,
+            to: tx.to,
+            data: tx.data,
+            chainId: tx.chainId,
+            from: tx.from,
+            nonce: txConfig.nonce,
+            gasPrice: Math.ceil(txConfig.gasPrice),
+            gasLimit: Math.ceil(txConfig.gasLimit),
+          }
+          tx = await this.wallet.sendTransaction(transaction)
+        }
+
+        logger.info(`Transaction pending`, { tx: tx })
+
+        const receipt = await this.ethereum.waitForTransaction(
+          tx?.hash,
+          3,
+          this.gasBumpTimeLimit,
+        )
+
+        if (receipt.status == 0) {
+          if (receipt.cumulativeGasUsed.toNumber() > txConfig.gasLimit * 0.9) {
+            throw indexerError(IndexerErrorCode.IE050)
+          }
+          throw indexerError(IndexerErrorCode.IE051)
+        }
+
+        logger.info(`Transaction successfully included in block`, {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          tx: tx?.hash,
+          receipt: receipt,
+        })
+        output = receipt
+        pending = false
+      } catch (error) {
+        txConfig = await this.updateTransactionConfig(logger, txConfig, error)
+        continue
+      }
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return output!
+  }
+
+  async updateTransactionConfig(
+    logger: Logger,
+    txConfig: Record<string, number>,
+    error: Error | IndexerError,
+  ): Promise<Record<string, number>> {
+    if (error instanceof IndexerError) {
+      if (error.code == 'IE050') {
```

Comment:
`error.code === IndexerErrorStatus.IE050` would be better, same with the check below.